From 3d2a33ba0b7b80757d4bc6535bbc2e6f89bcf0a3 Mon Sep 17 00:00:00 2001
From: Yuta Saito <kateinoigakukun@gmail.com>
Date: Tue, 20 Jun 2023 06:49:07 +0000
Subject: [PATCH] [wasm][cmake] HACK: Skip building swift-reflection-test for
 now

The reason why this change is needed is unclear to me...
---
 CMakeLists.txt                | 5 +++++
 stdlib/private/CMakeLists.txt | 2 +-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 54597829e72..8fb1b429b98 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -560,6 +560,11 @@ option(SWIFT_STDLIB_ENABLE_OBJC_INTEROP
        "Should stdlib be built with Obj-C interop."
        "${SWIFT_STDLIB_ENABLE_OBJC_INTEROP_default}")
 
+# FIXME(wasm) Reflection tests are temporalily disabled due to lack of linker features
+option(SWIFTWASM_DISABLE_REFLECTION_TEST
+    "Disable building swift-reflection-test for WebAssembly build"
+    FALSE)
+
 #
 # User-configurable experimental options.  Do not use in production builds.
 #
diff --git a/stdlib/private/CMakeLists.txt b/stdlib/private/CMakeLists.txt
index 05ef455b755..704d96f9248 100644
--- a/stdlib/private/CMakeLists.txt
+++ b/stdlib/private/CMakeLists.txt
@@ -28,7 +28,7 @@ if(SWIFT_BUILD_SDK_OVERLAY)
   endif()
   # Currently SwiftReflectionTest cannot be built on Windows, due to
   # dependencies on POSIX symbols
-  if (SWIFT_INCLUDE_TESTS AND (NOT CMAKE_SYSTEM_NAME STREQUAL "Windows"))
+  if (SWIFT_INCLUDE_TESTS AND (NOT CMAKE_SYSTEM_NAME STREQUAL "Windows") AND NOT SWIFTWASM_DISABLE_REFLECTION_TEST)
     add_subdirectory(SwiftReflectionTest)
   endif()
 endif()
-- 
2.40.0

